build/cmake/CMakeModules/AddZstdCompilationFlags.cmake

```diff
@@ -12,18 +12,19 @@ if (ZSTD_HAVE_CHECK_LINKER_FLAG)
 endif()
 
 function(EnableCompilerFlag _flag _C _CXX _LD)
+    string(REGEX REPLACE " " ";" flaglist "${_flag}")
```

Comment:
What about `separate_arguments()` ?

Reply:
In developing this fix I did encounter [`separate_arguments`](https://cmake.org/cmake/help/latest/command/separate_arguments.html) but it seems to do a whole lot of things that are unnecessary here. It is intended for parsing a program's command line arguments and handles quoting and escaping and other things which are not applicable here and would make it slower than just doing the single character replacement we want.

However, using just `SHELL:${_flag}`, and not splitting `${flag}` into a list, may be a better bet. Flags given as a list are assumed to be independent and subject to deduplication which we would not want here. For example, if `-z` `foo` were already in the flags and we test for `-z` `bar` cmake might remove the "duplicate" `-z` and end up with the invalid `-z` `foo` `bar`.

Reply:
Performance shouldn't be a concern here.
Rather I'm concerned by : 
- 2 keywords compilation flags, that should be considered as va single item: this <space> replacement will automatically separate both sides, this won't end well (and yes, I suspect everything works fine for now because there is no such 2-keywords flag at the moment the regex replace is invoked, but that's a recipe for future surprises)
- I wasn't aware of any implicit deduplication of a `cmake` list, it would indeed be a problem 
- The general "magic" of a regex replace, which doesn't explain the intent. I was initially considering a request for code comment to explain why this line is there. Then I discovered `separate_arguments()`, which would have the advantage of being a bit more clear about the intention, but possibly a bit less clear about the implementation.